### ISSUE TITLE
Using Bloazr WebAssembly Components in regular ASP.NET Web App

### ISSUE THREAD

**Issue by wutever0** [dotnet/aspnetcore#33183]
### Is your feature request related to a problem? Please describe.
I an new to Blazor and initially I imagined that one could create Blazor WebAssembly components and use them in a regular ASP.NET Web App. I understand that we can use Blazor Server components this way but it does not seem that we can use Blazor WebAssembly this way - Blazor WebAssembly seems to be an all or nothing deal.

### Describe the solution you'd like
Is my understanding correct?
Is this something that would be possible/feasible?
The benefit would be that the WebAssembly components would be run on the client.

**Comment by javiercn** [comment]
@wutever0 thanks for contacting us.

Can you clarify a bit what you are trying to do? 

It is totally possible to render a Blazor Webassembly component from an ASP.NET Core app. You need to make sure you have setup the Blazor app correctly on the ASP.NET Core app to serve everything that you need. We recommend starting with a Blazor webassembly hosted app for this (since all the setup is there) and changing the `index.html` on the client project for `_Host.cshtml` in the server project and `MapFallbackToFile` with `MapFallbackToPage`

**Comment by wutever0** [comment]
If I have a Blazor "counter" component that calls an IncrementCount function on click.
I would like to use that Blazor component in a traditional ASP.NET MVC application View and have the component created using WebAssembly so that the IncrementCount function runs on the client (as opposed to getting executed on the server) without having to turn my MVC application into a WebAssembly SPA.

Does that make sense?

**Comment by En3Tho** [comment]
@wutever0 There is a way to run Blazor like that, it's called Blazor server. As for mixing traditional views and Blazor ones I don't know, never tried.

**Comment by javiercn** [comment]
@wutever0 it's perfectly possible. You can use `RenderMode.Webassembly` instead of `Server` and in addition to that, you need to put the components in a webassembly project and perform the appropriate setup to bootstrap blazor webassembly on the browser from wherever you are rendering it on the server project.

**Comment by wutever0** [comment]
Thank you @javiercn..
So that would mean that I need to have two projects: first a web assembly project and then  my mvc application. Have the first as a dependency of the second and use the components in my mvc application?

Regarding the bootstrapping setup that you're referring to, I'm assuming that's all in the documentation? It was a bit hard to follow and I didn't find what I was looking for but if it's there I'll look again.

**Comment by javiercn** [comment]
@wutever0 that is correct.

I believe this is covered in the docs. However, I strongly recommend you start with a Blazor Webassembly hosted application (since that has an ASP.NET Core backend + the wasm setup) or that you create a Blazor wasm standalone project, add a reference to it from the asp.net core app and follow the docs (or mimic the code of a Blazor wasm hosted app)
